Modules/Filtering/DisplacementField/test/itkTimeVaryingVelocityFieldIntegrationImageFilterTest.cxx

```diff
@@ -72,6 +73,22 @@ itkTimeVaryingVelocityFieldIntegrationImageFilterTest(int argc, char * argv[])
   ITK_EXERCISE_BASIC_OBJECT_METHODS(integrator, TimeVaryingVelocityFieldIntegrationImageFilter, ImageToImageFilter);
 
 
+  auto velocityFieldInterpolator =
+    itk::VectorLinearInterpolateImageFunction<typename IntegratorType::TimeVaryingVelocityFieldType,
+                                              typename IntegratorType::ScalarType>::New();
+  integrator->SetVelocityFieldInterpolator(velocityFieldInterpolator);
+  ITK_TEST_SET_GET_VALUE(velocityFieldInterpolator, integrator->GetVelocityFieldInterpolator());
+
+  auto displacementFieldInterpolator =
+    itk::VectorLinearInterpolateImageFunction<typename IntegratorType::DisplacementFieldType,
+                                              typename IntegratorType::ScalarType>::New();
```

Comment:
Please remove these four `typename` keywords as well. Sorry!

Reply:
Done in [bf183be](https://github.com/InsightSoftwareConsortium/ITK/pull/4060/commits/bf183be9c4a49edec634ae6748f7e24e223cedee). Let's see what the dashboard says after merging, but plenty of cases are found across tests.

Reply:
FYI, from C++11 on (including C++14 and C++17), it appears _allowed_  to use `typename` keywords outside of a template. But it's still _unnecessary_ to do so, outside of a template.

Clang even introduced a warning saying:

> warning: 'typename' occurs outside of a template

Looking at https://godbolt.org/z/6YqjEPGG5 
```
struct Wrapper
{
    using Type = int;
};

typename Wrapper::Type i; // warning: 'typename' occurs outside of a template
```

So in fact it's really a matter of style, whether to use `typename` keywords in such cases.

Reply:
FYI the C++ standard tried to avoid the _need_ for such `typename` keywords as much as possible, looking at a paper like [Down with `typename`!](https://www.open-std.org/jtc1/sc22/wg21/docs/papers/2018/p0634r3.html)  There is also a follow-up paper, proposing to remove unnecessary `typename` keywords from the `std` library: [Down with `typename` in the Library!](https://open-std.org/JTC1/SC22/WG21/docs/papers/2020/p2150r0.html)

Reply:
Thanks for all he information Niels. Unfortunately, it is not about a particular coding habit, but about what the dashboard compilers report or have been reporting.

Reply:
> Unfortunately, it is not about a particular coding habit, but about what the dashboard compilers report or have been reporting.

If so, it could be some compiler bug, of course. A compiler should never require a `typename` keyword outside of a template. So if you encounter such a compile error again, please share a link to the specific dashboard page!

Reply:
Niels, the tests are full of such examples, even the files I modified contain examples; if CI and dashboard logs lived forever, you could probably navigate across similar coverage improvement PRs, compare push forced commits, dashboard states, etc. and see that in many cases the push force commits were done because a compiler complained about this.

Reply:
Do you mean this one? https://github.com/InsightSoftwareConsortium/ITK/compare/a612518c4fbec77e94112820183977e6aa6d18b4..a68d987c5b2703dde20a803c3036691668100c7a
```
  typename ConstantVelocityFieldTransformType::ConstantVelocityFieldInterpolatorPointer
    constantVelocityFieldInterpolator{};
```

That is actually inside a template.

Reply:
Just for the records, no new warnings have appeared after merging this, so not sure about past behavior, and `typename` instances outside templates. Anyways, thanks for the explanations @N-Dekker.